### PR TITLE
[NVFuser] Opinfos for extremal values

### LINKED ISSUE #75758: nvfuser opinfos for extremal values
### 🚀 The feature, motivation and pitch

Test supported ops at extremal values

we can copy the same strategy as test_binary_ufuncs.py

### Alternatives

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by davidberard98**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #76768
* __->__ #75917

Added slow tests for comparing the eager & fused outputs for given extremal inputs.

### CODE REVIEW

**Comment by eellison** [comment]
I think you can add a label `slow-tests` to trigger on your pr

**Comment by davidberard98** [comment]
@eellison I think right now it's ciflow/trunk, it was changed ~ 1 month ago

File: test/test_jit_cuda_fuser.py [hunk 1]
```diff
@@ -4559,12 +4568,64 @@ def test_nvfuser_correctness(self, device, dtype, op):
...
+                yield _get_extremal_sample(sample, val, dtype)
+
+        for sample in op.sample_inputs(device, dtype, requires_grad=False):
+
+            trace = create_traced_fn(self, op, cache_traced_fn=True)
+            trace(*clone_inputs((sample.input, *sample.args)), **sample.kwargs)
+            trace(*clone_inputs((sample.input, *sample.args)), **sample.kwargs)
+
+            for extremal_sample in _get_extremal_samples(sample, dtype):
+                ref = op(*clone_inputs((extremal_sample.input, *extremal_sample.args)), **extremal_sample.kwargs)
```
Comment (ngimel) [inline test/test_jit_cuda_fuser.py:4620]: you should catch errors here, many eager ops (like cholesky or other linalg functions) will error out. If eager function errors out, we shouldn't attempt to run traced function

File: test/test_jit_cuda_fuser.py [hunk 2]
```diff
@@ -4588,12 +4597,77 @@ def test_nvfuser_correctness(self, device, dtype, op):
...
         # each torch.jit.trace adds state to the _python_cu compilation unit
         # since this test traces a lot of functions, out-of-memory can occur
         # if the CU is not cleared.
         torch.jit._state._python_cu.drop_all_functions()
 
+    @slowTest
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    @ops(op_db, allowed_dtypes=(torch.float16, torch.bfloat16, torch.float32,
```
Comment (mruberry) [inline test/test_jit_cuda_fuser.py:4611]: What happens if nvFuser doesn't support the operator?

Comment (davidberard98) [reply test/test_jit_cuda_fuser.py:4611]: We'll still run the test, but the operator won't be fused by nvfuser (so it'll just run the usual eager implementation of the op)

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -14584,6 +14597,9 @@ def generate_std_var_kwargs(t: torch.Tensor, **kwargs):
                # lambda impl
                DecorateInfo(unittest.expectedFailure, "TestNormalizeOperators", "test_normalize_operator_exhaustive"),
                DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit'),
+               # 76571
```
Comment (jjsjann123) [inline torch/testing/_internal/common_methods_invocations.py:14600]: Thanks for tracking this.
Linking the issue #76571

**Review by jjsjann123 — APPROVED** [review]
LGTM

File: test/test_jit_cuda_fuser.py [hunk 4]
```diff
@@ -4588,12 +4597,76 @@ def test_nvfuser_correctness(self, device, dtype, op):
...
+            for extremal_sample in _get_extremal_samples(sample, dtype):
+                try:
+                    with freeze_rng_state():
+                        ref = variant(*clone_inputs((extremal_sample.input, *extremal_sample.args)),
+                                      **extremal_sample.kwargs)
+                except (torch._C._LinAlgError, RuntimeError, ValueError):
+                    # if eager errors out, then don't expect NVFuser to pass
+                    continue
+
+                with freeze_rng_state():
```
Comment (jjsjann123) [inline test/test_jit_cuda_fuser.py:4661]: nitpick: we've seen in a few cases where the test fails where there's no fusion happening.
I think those are still legit failures and should be reported by the tests. Does it make sense to explicitly report them by checking if the traced function has `CudaFusionGroup`?

Comment (davidberard98) [reply test/test_jit_cuda_fuser.py:4661]: To clarify - the idea here is to skip more checks, by skipping situations that don't have a `CudaFusionGroup` - is that right?

File: test/test_jit_cuda_fuser.py (hunk 2 above)
Comment (mruberry) [reply test/test_jit_cuda_fuser.py:4611]: That kinda seems excessive, right? Not the end of the world. In the future it'd be nice to know which operators are supported by nvFuser so we don't excessively test

Comment (davidberard98) [reply test/test_jit_cuda_fuser.py:4611]: Is there a good way to keep an up to date list of ops supported by NVFuser? I imagine that it would be pretty easy to forget to update the list when NVFuser adds new ops

Comment (mruberry) [reply test/test_jit_cuda_fuser.py:4611]: Yeah that's a good concern -- I don't know offhand

**Review by eellison — APPROVED** [review]
